File: mall-sellergoods-interface/src/main/java/com/marshal/sellergoods/service/SpecificationService.java
package com.marshal.sellergoods.service;


import com.marshal.pojo.TbSpecification;
import com.marshal.pojoext.Specification;
import com.marshal.util.ResponseData;
import java.util.List;
import java.util.Map;

public interface SpecificationService {
	public List<TbSpecification> findAll();

	List<Map> getOptionList();

	public ResponseData query(TbSpecification condition,int pageNum,int pageSize);

	public TbSpecification queryById(Long id);

	public void save(Specification Specification);

	public void delete(Long[] ids);
}

File: mall-manager-web/src/main/webapp/js/service/specificationService.js
// service层,与数据库交互
app.service("specificationService",function($http){
    //BASE crud
    this.query=function(pageNum,pageSize,condition){
        return $http.post("../specification/query?pageNum="+pageNum+"&pageSize="+pageSize,condition);
    }

    this.save=function (specification) {
        return $http.post("../specification/save",specification);
    }

    this.remove=function (idList) {
        return $http.get("../specification/delete?idList="+idList);
    }

    this.queryById=function (id) {
        return $http.get("../specification/queryById?id="+id);
    }
});